### PR TITLE
Add nginx module to fetch SVIDs

### PR DESCRIPTION
**Pull request by amartinezfayo**
- Add `ngx_http_fetch_spiffe_certs_module` nginx module to automatically fetch SVIDs.
- Include fixes to properly check SPIFFE ID of presented certificate.

### CODE REVIEW

File: ngx_http_fetch_spiffe_certs_module/ngx_http_fetch_spiffe_certs_module.c [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        ngx_conf_set_str_slot,
+        NGX_HTTP_SRV_CONF_OFFSET,
+        offsetof(ngx_http_ssl_spiffe_srv_conf_t, svid_bundle_file_path),
+        NULL },
+      ngx_null_command
+};
+
+static void * ngx_http_fetch_spiffe_certs_create_conf(ngx_conf_t *cf)
+{
+    ngx_http_ssl_spiffe_srv_conf_t  *scf;
```
Comment (MarcosDY) [inline ngx_http_fetch_spiffe_certs_module/ngx_http_fetch_spiffe_certs_module.c]: I think this struct must have a name like `ngx_http_fetch_spiffe_certs_create_conf_t`

File: ngx_http_fetch_spiffe_certs_module/ngx_http_fetch_spiffe_certs_module.c [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    scf = ngx_pcalloc(cf->pool, sizeof(ngx_http_ssl_spiffe_srv_conf_t));
+    if (scf == NULL) {
```
Comment (MarcosDY) [inline ngx_http_fetch_spiffe_certs_module/ngx_http_fetch_spiffe_certs_module.c]: why is this if necesary? if it is NULL you will still returning NULL at the end

File: ngx_http_fetch_spiffe_certs_module/ngx_http_fetch_spiffe_certs_module.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+func main() {}
+
+//export FetchSvids
+func FetchSvids(sockAddr *C.char,
+	svidFilePath *C.char,
+	svidKeyFilePath *C.char,
+	svidBundleFilePath *C.char) *C.char {
+
+	workloadClient, ctx, _, err := createGrpcClient(C.GoString(sockAddr))
+	m := &fetchSpiffeCertsModule{
```
Comment (MarcosDY) [inline ngx_http_fetch_spiffe_certs_module/ngx_http_fetch_spiffe_certs_module.go]: must no it go after error validation?

File: ngx_http_fetch_spiffe_certs_module/ngx_http_fetch_spiffe_certs_module.c (hunk 1 above)
Comment (amartinezfayo) [reply ngx_http_fetch_spiffe_certs_module/ngx_http_fetch_spiffe_certs_module.c]: Right, thanks.

File: ngx_http_fetch_spiffe_certs_module/ngx_http_fetch_spiffe_certs_module.go (hunk 3 above)
Comment (amartinezfayo) [reply ngx_http_fetch_spiffe_certs_module/ngx_http_fetch_spiffe_certs_module.go]: Yes :) Will fix it.

File: ngx_http_fetch_spiffe_certs_module/ngx_http_fetch_spiffe_certs_module.c (hunk 2 above)
Comment (amartinezfayo) [reply ngx_http_fetch_spiffe_certs_module/ngx_http_fetch_spiffe_certs_module.c]: It's not necessary. I'll fix it. Thanks.

**Pull request merged**